shadows/framework/src/main/java/org/robolectric/shadows/ShadowParcelFileDescriptor.java

```diff
@@ -28,6 +30,7 @@
 public class ShadowParcelFileDescriptor {
   // TODO: consider removing this shadow in favor of shadowing file operations at the libcore.os
   // level
+  private static final Map<Integer, Pair<String, Integer>> openedFileTable = new HashMap<>();
```

Comment:
Could we update map after fd changed for file?

Reply:
Yeah, I use a normal HashMap here, but I think the WeakHashMap<FileDescriptor, OpenedFileItem> should be used to avoid memory leak. But the performance can be traded off for each time  FileDescriptor should be searched from int fd.